eli5/sklearn/transform.py

```diff
@@ -21,15 +18,23 @@
 
 # Feature selection:
 
-@transform_feature_names.register(SelectorMixin)
-@transform_feature_names.register(RandomizedLogisticRegression)
-@transform_feature_names.register(RandomizedLasso)
```

Comment:
I had an idea for another approach here, but after some thought it looks worse than what is implemented here, still let me put it here:
- put imports at the top, set ``RandomizedLasso `` etc. to ``None`` if they fail to import
- make it possible to write something equivalent to ``@transform_feature_names.register(RandomizedLasso)`` which would do nothing if ``RandomizedLasso`` is ``None``. But it seems that we'd need a wrapper for that, something like ``@register(transform_feature_names, RandomizedLasso)``

Advantage is that the flow would be more natural (imports at the top, decorators above the function). Disadvantage is that it might be a bit confusing that ``RandomizedLasso`` can be ``None``, and also we would need to introduce the wrapper. So to me it seems that it would make sense only if we get a lot of similar conditional registers.

Reply:
Thanks for the suggestions @lopuhin. Looking at the [recently deprecated](https://scikit-learn.org/stable/modules/classes.html#recently-deprecated) in sklearn there doesn't seem to be too many removals of existing models. But I would certainly prefer your approach if we face more issues like this down the line.